jobs/kpi-forecasting/kpi_forecasting/models/base_forecast.py

```diff
@@ -70,26 +69,55 @@ def __post_init__(self) -> None:
             }
         )
 
-    def _fit(self) -> None:
-        """
-        Fit a forecasting model using `self.observed_df` that was generated using
-        Metric Hub data. This method should update `self.model`.
+    @abc.abstractmethod
+    def _fit(self, observed_df: pd.DataFrame) -> None:
```

Comment:
Not sure I follow why we're adding arguments to `_fit` and `_predict`. If the arguments are attributes of the class, why not just invoke the attributes in the method? This is what ends up happening in the `fit` and `predict` classes anyways.

I can see if we might want to generate the model and then fit/predict with different data or date ranges. In that case, I'd suggest modifying the arguments on the fit/predict methods so that they accept the same arguments, and default to the class's own attributes (see below).

Reply:
Yeah this is something I want to talk through because I keep going back and forth on it.  The idea is to "decouple" `BaseForecast` and it's children, so when you look at one of the child classes you minimize the stuff happening in the parent class you "need to know about".  I also thought it was a bit more intuitive having `_fit` be a function that you pass the training data to.  Finally, and this is more of a secondary consideration, it seemed a bit easier to test this way.

I like the idea of having the attributes be default args for `fit` and `predict`.  It does make the code more flexible!

Reply:
I keep updating the second sentence of the comment because nothing is sounding quite right, but at the end of the day I guess it just felt cleaner and more readable to me?

Reply:
Also whoops turns out you can't use self to set defaults like that